e2e/playwright/pay.spec.ts

```diff
@@ -0,0 +1,359 @@
+import { test, expect, type Page } from "@playwright/test";
+import { readFileSync } from "fs";
+import path from "path";
+import { gqlAdmin, insertTeam, insertTestUser } from "./utils";
+
+import team from "./assets/buckinhamshire.json";
+import ldcflow from "./assets/ldcFlow.json";
+import publishedFlows from "./assets/publishedFlowLDC.json";
+import operation from "./assets/operationLDC.json";
+
+test.setTimeout(90_000);
+
+const URL = "http://localhost:3000";
+
+const TEST_EMAIL = "test@test.com";
+const DEFAULT_WAIT_TIME = 300;
+
+test.describe("Payment flow", () => {
+  const USER_EMAIL = "test@test.com";
+
+  test.beforeAll(async () => {
+    await insertTestUser(USER_EMAIL);
+
+    await gqlAdmin(
+      `
+        mutation InsertTeams(
+          $teams: [teams_insert_input!]!, 
+        ) {
+          insert_teams(
+            objects: $teams,
+          ) {
+            affected_rows
+          }
+        }
+      `,
+      {
+        teams: [team],
+      }
+    );
+
+    await gqlAdmin(
+      `
+        mutation InsertFlows(
+          $flows: [flows_insert_input!]!,
+        ) {
+          insert_flows(
+            objects: $flows,
+          ) {
+            affected_rows
+          }
+        }
+      `,
+      {
+        flows: [ldcflow],
+      }
+    );
+
+    await gqlAdmin(
+      `
+        mutation InsertPublishedFlow(
+          $publishedFlows: [published_flows_insert_input!]!
+        ) {
+          insert_published_flows(
+            objects: $publishedFlows
+          ) {
+            affected_rows
+          }
+        }
+    `,
+      {
+        publishedFlows: publishedFlows,
+      }
+    );
+
+    await gqlAdmin(
+      `
+        mutation InsertOperations($operations: [operations_insert_input!]!) {
+          insert_operations(objects: $operations) {
+            affected_rows
+          }
+        }
+      `,
+      {
+        operations: [operation],
+      }
+    );
+  });
+
+  test.afterAll(async () => {
+    await gqlAdmin(`
+      mutation CleanDatabase {
+        delete_session_backups(where: {}){
+          affected_rows
+        }
+        delete_analytics_logs(where: {}){
+          affected_rows
+        }
+        delete_analytics(where: {}){
+          affected_rows
+        }
+        delete_published_flows(where: {}){
+          affected_rows
+        }
+        delete_operations(where: {}){
+          affected_rows
+        }
+        delete_flows(where: {}) {
+          affected_rows
+        }
+        delete_teams(where: {}) {
+          affected_rows
+        }
+      }
+    `);
+  });
+
+  test.skip("Should pay and send the application", async ({ page }) => {
```

Comment:
The `unpublished` flow was skipped from the tests. In order to test it correctly we'll have to add more data to the tests:

* insert every flow that LDC uses as a portal
* insert the operation, team, and published_flow for every inserted flow

Let me know what you guys think.

Reply:
I think only testing `/preview` makes sense for now, as our goal is to test applicant Pay journeys & applicants should never be on `/unpublished` routes !